x-pack/test/security_solution_cypress/cypress/e2e/detection_response/rule_management/related_integrations/related_integrations.cy.ts

```diff
@@ -212,42 +208,26 @@ describe('Related integrations', { tags: ['@ess', '@serverless', '@brokenInServe
         });
       });
 
-      it('the alerts generated should have a "kibana.alert.rule.parameters.related_integrations" field containing the integrations', () => {
-        const RELATED_INTEGRATION_FIELD = 'kibana.alert.rule.parameters.related_integrations';
+      const RELATED_INTEGRATION_FIELD = 'kibana.alert.rule.parameters.related_integrations';
 
+      it(`the alerts generated should have a "${RELATED_INTEGRATION_FIELD}" field containing the integrations`, () => {
         deleteDataStream(DATA_STREAM_NAME);
         createDocument(DATA_STREAM_NAME, generateEvent());
 
-        waitForPageToBeLoaded(PREBUILT_RULE_NAME);
         enablesRule();
         waitForAlertsToPopulate();
-        expandFirstAlert();
-        openTable();
-        filterBy(RELATED_INTEGRATION_FIELD);
-
-        cy.get(FIELD(RELATED_INTEGRATION_FIELD))
-          .invoke('text')
-          .then((stringValue) => {
-            // Integrations are displayed in the flyout as a string with a format like so:
-            // '{"package":"aws","version":"1.17.0","integration":"unknown"}{"package":"mock","version":"1.1.0"}{"package":"system","version":"1.17.0"}'
-            // We need to parse it to an array of valid objects before we can compare it to the expected value
-            // Otherwise, the test might fail because of the order of the properties in the objects in the string
-            const jsonStringArray = stringValue.split('}{');
-
-            const validJsonStringArray = createValidJsonStringArray(jsonStringArray);
-
-            const parsedIntegrations = validJsonStringArray.map((jsonString) =>
-              JSON.parse(jsonString)
-            );
-
-            RULE_RELATED_INTEGRATIONS.forEach((integration) => {
-              expect(parsedIntegrations).to.deep.include({
-                package: integration.package,
-                version: integration.version,
-                ...(integration.integration ? { integration: integration.integration } : {}),
-              });
-            });
+
+        fetchRuleAlerts({
+          ruleId: 'rule_1',
+          fields: [RELATED_INTEGRATION_FIELD],
+          size: 1,
+        }).then((alertsResponse) => {
+          expect(alertsResponse.body.hits.hits[0].fields).to.deep.equal({
+            [RELATED_INTEGRATION_FIELD]: RULE_RELATED_INTEGRATIONS.map((x) =>
+              omit(x, ['installed', 'enabled'])
+            ),
```

Comment:
Thank you for this improvement, it looks much more reliable now 👍 

Should this PR close https://github.com/elastic/kibana/issues/165504 or do you think we should move this test to integration tests later?

Reply:
Yes, it should.